src/foam/nanos/auth/User.js

```diff
@@ -229,36 +228,47 @@ foam.CLASS({
       section: 'administrative'
     },
     {
-      class: 'FObjectProperty',
-      of: 'foam.nanos.auth.Phone',
-      name: 'phone',
+      class: 'String',
+      name: 'phoneNumber',
       documentation: 'Personal phone number.',
-      factory: function() {
-        return this.Phone.create();
-      },
-      view: { class: 'foam.u2.detail.VerticalDetailView' },
+//      validationPredicates: [
```

Comment:
We should just refine `foam.core.PhoneNumber` to add the validation predicates there, then we don't need to repeat it.

Reply:
Question, when you add a validationPredicate to a class, do you refer to it by 'this'?
Example:
foam.CLASS({
  package: 'foam.core',
  name: 'PhoneNumber',
  extends: 'String',
  label: 'Phone number'
});


foam.CLASS({
  package: 'foam.core',
  name: 'FormattedPhoneNumber',
  extends: 'PhoneNumber',
  label: 'Formatted Phone number',
  validationPredicates: [
    {
      args: ['this'],
      predicateFactory: function(e) {
        return e.OR(
                      e.REG_EXP(
                                    e,
                                    /^(?:\+?1[-.●]?)?\(?([0-9]{3})\)?[-.●]?([0-9]{3})[-.●]?([0-9]{4})$/),
                      foam.mlang.predicate.False.create()
                    );
      },
      errorString: 'Please enter a valid phone number'
    }
  ]
});

Reply:
Check out https://github.com/foam-framework/foam2/blob/master/src/foam/core/Validation.js#L309-L344 for an example.